playbooks/common/openshift-master/config.yml

```diff
@@ -337,12 +361,54 @@
     with_items: openshift_master_named_certificates
     when: named_certs_specified | bool
 
-- name: Configure master instances
-  hosts: oo_masters_to_config
+- name: Configure first master instance
```

Comment:
> You could do this by using set_fact to conditionally set it based on whether the host is the "first master". The template in the set_fact task will be evaluated at run time, and as long as the play is still serialized we would know that we have already run on the first master.
>
> Also, since the act of running openshift_facts updates the local facts for the host, by the time the subsequent masters are run, the fact variable would already be set.

I was already mid-way on iterating on this setting of openshift_version and eliminating the duplicated blocks. The current implementation has a play for first master that just determines the initial openshift_version if possible in the playbook (depending on what you set in inventory), then calls the openshift_docker role (which is likely due for a rename soon), which will make absolutely sure openshift_version is then set to something fully specified. (i.e. it will look up latest if the inventory doesn't specify anything, and it will convert 3.2 to 3.2.0.44) When this play completes, we *know* openshift_version is set.

Then I just run another play for the other masters that sets openshift_version to openshift.common.version if it exists (i.e. protect what's already installed), otherwise the value calculated above for first master.

I think this should get us there, most of that code is live in openshift-master/config.yml if you'd like to take a look. (haven't removed the duplication yet but I could very soon) 



>> I also just noticed the subsequent masters are not properly protecting their installed versions, i.e. if first master somehow got a newer version, a re-run of config playbook would upgrade the subsequent masters. Adding that to TODO as well.

> I'm not completely convinced they should, either. if the config playbooks are intended to keep the cluster configured and in sync, the version from the first master should be the starting point, I believe.

Hmm, so you're saying for subsequent masters, always key them off the current version from the first master? I was operating on the "protect what is installed unless explicitly upgrading" goal, but that may have been node specific.

I assume for nodes, we still will preserve current version unless upgrading though correct?